### DISCUSSION TITLE
No code when using my own search command “npm run swizzle @docusaurus/theme-classic SearchBar”

### DISCUSSION THREAD

**Discussion by KulkiratSinghSonrai** [facebook/docusaurus#4104] [Q&A]
Hello everyone. I am new to Docusaurus and I am trying to use my own search as given in the documentation using "npm run swizzle @docusaurus/theme-classic SearchBar " command. After running the command, I do not see any code in the SearchBar.js file. I saw in an example that after running the command, there were bunch of lines of code which we can modify to add functionality in the project but when I run the command, I get just a single line of code. Can anyone let me know what I am doing wrong ? 

![image](https://user-images.githubusercontent.com/77124996/105787054-8c429f00-5f54-11eb-8408-d59b3876ddb1.png)

**Comment by slorber** [comment]
Hi,

By default the classic theme does not provide any searchbar, that's why you end-up with this "no-op" component.

Run `npm run swizzle @docusaurus/theme-search-algolia SearchBar` instead so you can customize the algolia searchbar component.

If you don't want to use Algolia, you'll actually have to build your own searchbar component from an empty file.

**Reply by dandv** [reply]
Swizzling that Algolia search bar doesn't work right off the bat. I've run into error after error, ended up having to add `themes: ['@docusaurus/theme-search-algolia']` to the config and these dummy fields to `themeConfig` in `docusaurus.config.js`,

```
        algolia: {
            appId: 'foo',
            apiKey: 'bar',
            indexName: 'baz',
        }
```

but finally got stuck here:

```
npm run swizzle @docusaurus/theme-search-algolia SearchBar

> my-website@0.0.0 swizzle
> docusaurus swizzle @docusaurus/theme-search-algolia SearchBar

[ERROR] Error: Plugin "docusaurus-theme-search-algolia" is used 2 times with ID "default".
To use the same plugin multiple times on a Docusaurus site, you need to assign a unique ID to each plugin instance.
The plugin ID is "default" by default. It's possible that the preset you are using already includes a plugin instance, in which case you either want to disable the plugin in the preset (to use a single instance), or assign another ID to your extra plugin instance (to use multiple instances).
```

**Reply by slorber** [reply]
> Swizzling that Algolia search bar doesn't work right off the bat.

Please read the doc and follow the instructions more sequentially and carefully: https://docusaurus.io/docs/next/search#using-algolia-docsearch 

It has never been said that swizzling a plugin that you didn't use/install in the first place would work 😅 

![CleanShot 2022-12-15 at 13 17 22@2x](https://user-images.githubusercontent.com/749374/207857399-3d56bf07-4eb6-4474-b45a-d9ac1e4c2dbb.png)

>  I've run into error after error, ended up having to add themes: ['@docusaurus/theme-search-algolia'] to the config and these dummy fields to themeConfig in docusaurus.config.js,

Our doc doesn't say to add this entry to themes

> Docusaurus' own @docusaurus/preset-classic supports Algolia DocSearch integration. If you use the classic preset, no additional installation is needed.
> Then, add an algolia field in your themeConfig.

That's all, just `themeConfig.algolia`, no `themes: []` to add because it's already part of the preset. 

If you do that, then the theme gets registered twice (by yourself + by the preset)

And we do support multiple instances of a plugin/theme, but you have to assign an explicit id to each of them.

See https://docusaurus.io/docs/next/using-plugins#multi-instance-plugins-and-plugin-ids

---

Not sure how to improve your experience and make the error messages more obvious. If you have suggestions, please let us know or open a PR.

**Reply by dandv** [reply]
> It has never been said that swizzling a plugin that you didn't use/install in the first place would work 😅

I never expected that :) But I did use the vague word "work", so let me detail.

> Not sure how to improve your experience and make the error messages more obvious. If you have suggestions, please let us know or open a PR.

The issue is that I did follow the instructions, albeit perhaps too literally. The instruction was to run

    npm run swizzle @docusaurus/theme-search-algolia SearchBar

I expected that to produce an empty component, without errors. The problem was that it produced this error:

> [ERROR] Theme @docusaurus/theme-search-algolia not found. Themes available for swizzle:
> - @docusaurus/theme-classic
> - @docusaurus/plugin-debug

That led me to add `themes: ['@docusaurus/theme-search-algolia']`, which you pointed out that I shouldn't do, and instead I should only add the three config values. That could be spelled out in the instructions, so I created [this PR](https://github.com/facebook/docusaurus/pull/8444). I do get an Unsafe warning when swizzling, and not sure yet what to do at that point, but I'll keep reading the docs. Thanks!
